src/Symfony/Component/Routing/Requirement/EnumRequirement.php

```diff
@@ -22,18 +22,26 @@ final class EnumRequirement implements \Stringable
 
     /**
      * @template T of \BackedEnum
+     *
      * @param class-string<T> $enum
-     * @param T ...$cases
+     * @param T|list<T>       $cases
      */
-    public function __construct(string $enum, \BackedEnum ...$cases)
+    public function __construct(string $enum, \BackedEnum|array $cases = [])
```

Comment:
My proposal was to have only one argument ;)
(Please squash the commits btw)

Reply:
@wouterj @stof do you agree? I misunderstood Nicolas' proposal.